pkg/lib/config/configsource/kubernetes.go

```diff
@@ -190,19 +191,39 @@ func (k *Kubernetes) Close() error {
 }
 
 func (k *Kubernetes) AllAppIDs() ([]string, error) {
-	appIDs := k.appIDs.Load().([]string)
+	// FIXME(k8s): remove this after introducing proper authz
+	ingresses, err := k.Client.NetworkingV1beta1().Ingresses(k.Namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	apps := map[string]struct{}{}
+	for _, ingress := range ingresses.Items {
+		if appID, ok := ingress.Labels[LabelAppID]; ok && appID != "" {
+			apps[appID] = struct{}{}
+		}
+	}
+
+	var appIDs []string
+	for appID := range apps {
+		appIDs = append(appIDs, appID)
+	}
+	sort.Strings(appIDs)
+
 	return appIDs, nil
 }
 
 func (k *Kubernetes) ResolveAppID(r *http.Request) (string, error) {
 	host := httputil.GetHost(r, bool(k.TrustProxy))
-	hostMap := k.hostMap.Load().(map[string]string)
+	if h, _, err := net.SplitHostPort(host); err == nil {
```

Comment:
Any particular reason why do we need to ignore the port?

Reply:
Kubernetes Ingress does not support routing on request host port.